packages/account/src/providers/transaction-request/helpers.ts

```diff
@@ -93,3 +96,38 @@ export const cacheRequestInputsResourcesFromOwner = (
       messages: [],
     } as Required<ExcludeResourcesOption>
   );
+
+export const getBurnableAssetCount = (opts: {
+  inputs: TransactionRequestInput[];
+  outputs: TransactionRequestOutput[];
+}) => {
+  const { inputs, outputs } = opts;
+  const coinInputs = new Set(inputs.filter(isRequestInputCoin).map((input) => input.assetId));
+  const changeOutputs = new Set(
+    outputs.filter((output) => output.type === OutputType.Change).map((output) => output.assetId)
+  );
+  const difference = new Set([...coinInputs].filter((x) => !changeOutputs.has(x)));
+  return difference.size;
+};
+
+export const validateTransactionForAssetBurn = (
+  tx: TransactionRequest,
+  enableAssetBurn: boolean = false
```

Comment:
```suggestion
  enableAssetBurn: boolean | undefined
```

Reply:
@nedsalk Why not:

```suggestion
  enableAssetBurn?: boolean
```
🤣

Reply:
IMO setting an explicit default here is clearer to read.